control/grpc.py

```diff
@@ -122,6 +124,19 @@ def get_hosts_with_dhchap_key(self, subsys):
             return self.host_dhchap_key[subsys]
         return {}
 
+    def add_host_nqn(self, subsys, hostnqn):
+        self.host_nqn[subsys].add(hostnqn)
+
+    def remove_host_nqn(self, subsys, hostnqn):
+        if not subsys in self.host_nqn:
+            return
+        self.host_nqn[subsys].discard(hostnqn)
+
+    def get_host_count(self, subsys):
+        if not subsys in self.host_nqn:
+            return 0
+        return len(self.host_nqn[subsys])
```

Comment:
```suggestion
        return len(self.host_nqn.get(subsys, set()))
```

Great [point](https://github.com/ceph/ceph-nvmeof/pull/952/files#r1844964974)! Using get ensures that the defaultdict is not modified if subsys does not exist. wdyt?

Reply:
I prefer to keep the "if" as it is much clearer to understand. I don't try to save one line of code.